api/src/services/users.js

```diff
@@ -518,11 +518,15 @@ module.exports = {
   */
   createAdmin: userCtx => {
     return validateUser(createID(userCtx.name))
-      .catch(() => {
-        const data = { username: userCtx.name, roles: ['admin'] };
-        return validateNewUsername(userCtx.name)
-          .then(() => createUser(data, {}))
-          .then(() => createUserSettings(data, {}));
+      .catch(err => {
+        if(err.status && err.status === 404) {
+          const data = { username: userCtx.name, roles: ['admin'] };
+          return validateNewUsername(userCtx.name)
+            .then(() => createUser(data, {}))
+            .then(() => createUserSettings(data, {}));
+        } else {
+          return Promise.reject(err);
```

Comment:
I _think_ your intention in the condition was to null check `err`, right? Otherwise `err.status && err.status === 404` is kind of redundant. We can also make a slight improvement and remove the unnecessary else
```suggestion
        if (err && err.status === 404) {
          const data = { username: userCtx.name, roles: ['admin'] };
          return validateNewUsername(userCtx.name)
            .then(() => createUser(data, {}))
            .then(() => createUserSettings(data, {}));
        }
        return Promise.reject(err);
```